File: codingsample/202008/20200824/Rains_of_Reason_1_arrayReplace.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h> 

 typedef struct arr_integer {
   int size;
   int *arr;
 } arr_integer;

 arr_integer alloc_arr_integer(int len) {
   arr_integer a = {len, len > 0 ? malloc(sizeof(int) * len) : NULL};
   return a;
 }
arr_integer insertEM(arr_integer a, int *b){
	int len = a.size,i;
	for(i=0;i<len;i++) a.arr[i] = b[i];
	return a;
}

arr_integer arrayReplace(arr_integer inputArray, int elemToReplace, int substitutionElem) {
	int i;
	for(i=0;i<inputArray.size;i++) if(inputArray.arr[i]==elemToReplace) inputArray.arr[i]=substitutionElem;
	return inputArray;
}


int main() {
	arr_integer input = alloc_arr_integer(3);
	int a[3] = {1, 2, 1};
	insertEM(input,a);
	int i;
	arrayReplace(input,1,3);
	for(i=0;i<input.size;i++) printf("%d\n",input.arr[i]);
	
}

File: codingsample/202008/20200824/Rains_of_Reason_1_arrayReplace.java
package N6_Rains_of_Reason;

import java.util.Arrays;

public class Rains_of_Reason_1_arrayReplace {
	static int[] arrayReplace(int[] inputArray, int elemToReplace, int substitutionElem) {
		for(int i=0;i<inputArray.length;i++) if(inputArray[i]==elemToReplace) inputArray[i] = substitutionElem;
		return inputArray;
	}
	
	static int[] arrayReplace2(int[] inputArray, int elemToReplace, int substitutionElem) {
		return Arrays.stream(inputArray).map(i -> i == elemToReplace ? substitutionElem : i).toArray();
	}
	
	public static void main(String[] args) {
		int[] a = {1,2,1};
		for(int i=0;i<a.length;i++) System.out.print(arrayReplace(a,1,3)[i]+" ");
		System.out.println();
		for(int i=0;i<a.length;i++) System.out.print(arrayReplace2(a,1,3)[i]+" ");
		System.out.println();
	}
	
}
